language/go/generate.go

```diff
@@ -79,7 +79,8 @@ func (gl *goLang) GenerateRules(args language.GenerateArgs) language.GenerateRes
 	// to any .proto files present.
 	regularFiles := append([]string{}, args.RegularFiles...)
 	genFiles := append([]string{}, args.GenFiles...)
-	if !pcMode.ShouldIncludePregeneratedFiles() {
+	gc := getGoConfig(c)
```

Comment:
It might be simpler to change `getProtoMode` to return `proto.DisableMode` when `gc.goProto` is false. That would cause everything to behave as if proto rule generation were disabled, even though there are generated proto rules.